lambdas/url_access_patterns/lambda_for_getHash/index.js

```diff
@@ -11,40 +11,43 @@ function response(statusCode, message) {
   };
 }
 
-exports.getHash = async (event) =>{
-    try{
-        var params={
-            TableName:"V-Transfer",
-            IndexName:"FIND_FILE_BY_URLID",
-            KeyConditionExpression: "GS1_PK= :pk",
-            ExpressionAttributeValues:{
-                ':pk':`${event.path.url_id}`,                
-            },                                    
-        }        
-        var url_data = await dynamo.query(params).promise()  
-        if(url_data.Items[0]==undefined) return response(404,"Provided URL is invalid"); 
-        url_data=url_data.Items[0];
-        if(url_data.visible===false)return response(403,"Provided URL is not accessible")
-        if(url_data.clicks_left<=0)return response(403,"Provided URL is not accessible")
-    }
-    catch(err){
-        return response(err.statusCode,err.message);
-    }
-    try{
-        await dynamo.update({
-            TableName: "V-Transfer",
-            Key: {
-                PK: `${url_data.PK}`,
-                SK: `${url_data.SK}`
-            },
-            UpdateExpression: "add clicks_left :val",
-            ExpressionAttributeValues: {
-                ":val": -1
-            }
-        }).promise();
-        return response(200,url_data.hash);
-    }
-    catch(err){
-        return response(err.statusCode,err.message);
-    }
-}
\ No newline at end of file
+exports.handler = async (event) => {
+  try {
+    var params = {
+      TableName: "V-Transfer",
+      IndexName: "FIND_FILE_BY_URLID",
+      KeyConditionExpression: "GS1_PK= :pk",
+      ExpressionAttributeValues: {
+        ":pk": `${event.path.urlId}`,
```

Comment:
why use wildcard if you aren't adding any suffix/prefix?

I think `":pk": event.path.urlId`, would work

Reply:
Just in case, following general pattern

Reply:
it's not a general pattern, use the feature only if it solves the purpose.